rclcpp/src/rclcpp/node_interfaces/node_topics.cpp

```diff
@@ -43,23 +43,25 @@ NodeTopics::create_publisher(
 
   // Setup intra process publishing if requested.
   if (use_intra_process) {
-    auto context = node_base_->get_context();
-    // Get the intra process manager instance for this context.
-    auto ipm = context->get_sub_context<rclcpp::intra_process_manager::IntraProcessManager>();
-    // Register the publisher with the intra process manager.
+    // Check if the QoS allows to use intra-process
     if (publisher_options.qos.history == RMW_QOS_POLICY_HISTORY_KEEP_ALL) {
       throw std::invalid_argument(
               "intraprocess communication is not allowed with keep all history qos policy");
     }
-    if (publisher_options.qos.depth == 0) {
```

Comment:
Actually this situation would result in throwing an error later, when the intra-process buffer is created.
Restoring this check

Reply:
In the current implementation the buffer is on the subscription side, so this is not needed.
However, after this PR I can add support for transient local so this will become a condition relevant only when IPC is on and durability is not volatile.